src/main/java/com/project/springbootproject/model/ShoppingCart.java

```diff
@@ -0,0 +1,41 @@
+package com.project.springbootproject.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@SQLDelete(sql = "UPDATE shoppingcarts SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
+@Entity
+@Data
+@Table(name = "shoppingcarts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NonNull
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    @EqualsAndHashCode.Exclude
+    private User user;
+    @OneToMany(mappedBy = "shoppingCart", cascade = CascadeType.ALL)
+    @EqualsAndHashCode.Exclude
+    private Set<CartItem> cartItems;
+    private boolean isDeleted = false;
+
+    public ShoppingCart() {
+
+    }
```

Comment:
```suggestion
```
will be created by default

Reply:
without default constructor compilation error